docs/source/examples/checkpointer.rst

```diff
@@ -0,0 +1,453 @@
+:ref:`LoRA Finetuning Tutorial <lora_finetune_label>`.
+
+The primary difference between the two use cases is when you want to resume training
+from a checkpoint. In this case, the checkpointer needs access to both the initial frozen
+base model weights as well as the learnt adapter weights. The config for this scenario
+looks something like this:
+
+
+.. code-block:: yaml
+
+    checkpointer:
+
+        # checkpointer to use
+        _component_: torchtune.utils.FullModelHFCheckpointer
+
+        # directory with the checkpoint files
+        # this should match the output_dir above
+        checkpoint_dir: <checkpoint_dir>
+
+        # checkpoint files. This is the ORIGINAL frozen checkpoint
+        # and NOT the merged checkpoint output during training
+        checkpoint_files: [
+            pytorch_model-00001-of-00002.bin,
+            pytorch_model-00002-of-00002.bin,
+        ]
+
+        # this refers to the adapter weights learnt during training
+        adapter_checkpoint: adapter_0.pt
+
+        # the file with the checkpoint state
+        recipe_checkpoint: recipe_state.pt
+
+        # dir for saving the output checkpoints. Usually set
+        # to be the same as checkpoint_dir
+        output_dir: <checkpoint_dir>
+
+        # model_type which specifies how to convert the state_dict
+        # into a format which TorchTune understands
+        model_type: LLAMA2
+
+    # set to True if restarting training
+    resume_from_checkpoint: True
+
+
+Putting it all Together
+-----------------------
+
+Let's now take the concepts above and see how we can use an input checkpoint, run
+finetuning and then inspect model quality by running some generations.
+
+1. Run LoRA finetuning on a single GPU using the default config
+
+.. code-block:: bash
+
+    tune run lora_finetune_single_device \
+    --config llama2/7B_lora_single_device
+
+    # output at the end of the training
+    [_checkpointer.py:456] Model checkpoint of size 9.98 GB saved to /tmp/Llama-2-7b-hf/hf_model_0001_0.pt
+    [_checkpointer.py:456] Model checkpoint of size 3.50 GB saved to /tmp/Llama-2-7b-hf/hf_model_0002_0.pt
+    [_checkpointer.py:467] Adapter checkpoint of size 0.01 GB saved to /tmp/Llama-2-7b-hf/adapter_0.pt
+
+
+2. Update the generation recipe with these checkpoints. The new config has the following changes to
+``generate.yaml``
+
+.. code-block:: yaml
+
+    checkpointer:
+        _component_: torchtune.utils.FullModelHFCheckpointer
+        checkpoint_dir: /tmp/Llama-2-7b-hf/
+        checkpoint_files: [
+            hf_model_0001_0.pt,
+            hf_model_0002_0.pt
+        ]
+        output_dir: /tmp/Llama-2-7b-hf/
+        model_type: LLAMA2
+
+3. Run the generation recipe with the new config
+
+.. code-block:: bash
+
+    tune run generate --config generate
+
+    # output from the generation
+    [generate.py:68] Model is initialized with precision torch.bfloat16.
+    [generate.py:92] Welcome to the 'Alternative' Treatments and Therapies Forum
```

Comment:
nit: without understanding the generation recipe I have no idea what's going on here (even with understanding it I kinda have no idea what's going on here 😅)

Reply:
oh you mean in the logs - hmm, let me take a closer look